src/kernel/subtract_indexed_outputs/cpu.cc

```diff
@@ -46,24 +48,29 @@ void cpu(Index n_labels, Index n_outputs, Scalar val_, const int64_t* labels_,
     Y dst_val{0.0};
     for(Index i = 0; i < n_outputs; ++i)
     {
-        dst_val = static_cast<Y>(dst_[labels[i] + i*n_labels]);
-        dst_[labels[i] + i*n_labels] = static_cast<T>(dst_val - val);
+        if (labels[i] == ignore_index)
+        {
+            std::memset(dst_ + i*n_labels, 0, n_labels * sizeof(T));
```

Comment:
If a label shall be ignored, its gradient shall be untouched. In the beginning all gradients are zeroed and some operations might update them. If `subtract_indexed_outputs` is called after another function that actually updates gradient, it will destroy already accumulated gradient. And that is a bug. There shall be no update in case a label is to be ignored.